### PR TITLE
Missing migration types

### PR DESCRIPTION
**Pull request by philkra**
added and validated the endpoints:
* `migration.get_feature_upgrade_status`
* `migration.post_feature_upgrade`

### CODE REVIEW

File: specification/migration/get_feature_upgrade_status/GetFeatureUpgradeStatusResponse.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Indices, VersionString } from '@_types/common'
+
+export class Response {
+  body: {
+    features: MigrationFeature[]
+    migration_status: string
```
Comment (sethmlarson) [inline specification/migration/get_feature_upgrade_status/GetFeatureUpgradeStatusResponse.ts:25]: Looks like this field is based on an enum, should we use one here too?

https://github.com/elastic/elasticsearch/blob/26c86871fc091900952e88e252c36fbfedf8d5fa/server/src/main/java/org/elasticsearch/action/admin/cluster/migration/GetFeatureUpgradeStatusResponse.java#L106

File: specification/migration/get_feature_upgrade_status/GetFeatureUpgradeStatusResponse.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  }
+}
+
+export class MigrationFeature {
+  feature_name: string
+  minimum_index_version: VersionString
+  migration_status: string
+  indices: Indices
```
Comment (sethmlarson) [inline specification/migration/get_feature_upgrade_status/GetFeatureUpgradeStatusResponse.ts:33]: From this XContent builder https://github.com/elastic/elasticsearch/blob/26c86871fc091900952e88e252c36fbfedf8d5fa/server/src/main/java/org/elasticsearch/action/admin/cluster/migration/GetFeatureUpgradeStatusResponse.java#L298 I think the structure for `indices` is:

```ts
export class MigrationFeatureIndexInfo {
  index: IndexName
  version: VersionString
  failure_cause?: ErrorCause
}
```

and then `MigrationFeature.indices` can be `MigrationFeatureIndexInfo[]`.

**Review by sethmlarson — CHANGES_REQUESTED** [review]
Two comments:

File: specification/migration/get_feature_upgrade_status/GetFeatureUpgradeStatusResponse.ts (hunk 2 above)
Comment (philkra) [reply specification/migration/get_feature_upgrade_status/GetFeatureUpgradeStatusResponse.ts]: Thank you for elaborating, addressed the feedback in https://github.com/elastic/elasticsearch-specification/pull/1393/commits/794f0f4f2b0903dea03ebee6f794163aad6c46bf

**Review by sethmlarson — APPROVED** [review]

**Pull request merged**